spring-cloud-app-starter-metadata-maven-plugin/src/main/java/org/springframework/cloud/stream/app/documentation/plugin/MetadataAggregationMojo.java

```diff
@@ -267,10 +341,89 @@ private String toJson(ConfigurationMetadata metadata) throws IOException {
 		} catch (Exception e) {
 			throw new MojoExecutionException("Exception trying to read metadata from dependencies of project", e);
 		}
+		if (!inboundPorts.isEmpty()) {
+			visible.put("configuration-properties.inbound-ports", StringUtils.arrayToCommaDelimitedString(inboundPorts.toArray(new String[0])));
+		}
+		if (!outboundPorts.isEmpty()) {
+			visible.put("configuration-properties.outbound-ports", StringUtils.arrayToCommaDelimitedString(outboundPorts.toArray(new String[0])));
+		}
 		return visible;
 	}
 
 
+	String[] gatherInboundPorts() throws MojoExecutionException {
+		List<String> inboundPorts = new ArrayList<>();
+		try {
+			for (String path : mavenProject.getRuntimeClasspathElements()) {
+				if (Files.isDirectory(Paths.get(path))) {
+					File dir = new File(path);
+					for (File file : dir.listFiles()) {
+						if (file.isFile() && (file.getName().endsWith(".yaml") || file.getName().endsWith(".yml")
+								|| file.getName().endsWith(".properties"))) {
+							if (file.canRead()) {
+								Properties properties = new Properties();
+								try (InputStream is = new FileInputStream(file)) {
+									properties.load(is);
+									if (properties.containsKey("spring.cloud.function.definition")) {
+										String functionDefinition = properties
+												.getProperty("spring.cloud.function.definition");
+										for (Object propertyKey : properties.keySet()) {
+											if (((String) propertyKey).startsWith(
+													"spring.cloud.stream.function.bindings." + functionDefinition
+															+ ".in-")) {
+												inboundPorts.add(properties.getProperty((String) propertyKey));
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		} catch (Exception e) {
+			throw new MojoExecutionException("Exception trying to read metadata from dependencies of project", e);
+		}
+		return inboundPorts.toArray(new String[0]);
+	}
+
+	String[] gatherOutboundPorts() throws MojoExecutionException {
+		List<String> outboundPorts = new ArrayList<>();
+		try {
+			for (String path : mavenProject.getRuntimeClasspathElements()) {
+				if (Files.isDirectory(Paths.get(path))) {
+					File dir = new File(path);
+					for (File file : dir.listFiles()) {
+						if (file.isFile() && (file.getName().endsWith(".yaml") || file.getName().endsWith(".yml")
```

Comment:
While the paring logic below 
```
spring.cloud.stream.function.bindings.${spring.cloud.function.definition|spring.cloud.stream.function.definition}-in-xxx
spring.cloud.stream.function.bindings.${spring.cloud.function.definition|spring.cloud.stream.function.definition}-out-xxx
```
would work for java properties files it will fail short to handle hierarchical YAML properties. E.g. 
```  
spring:
  cloud:
    ....
````
I guess we need to use a dedicated YML parsing utilities to handle yuml and and yml files.